content/en/docs/concepts/workloads/controllers/replicaset.md

```diff
@@ -88,66 +102,114 @@ Events:
+frontend-pxj4r   1/1     Running   0          5s
+pod1             1/1     Running   0          13s
+pod2             1/1     Running   0          13s
+```
 
-The `.spec.selector` field is a [label selector](/docs/concepts/overview/working-with-objects/labels/). A ReplicaSet
-manages all the pods with labels that match the selector. It does not distinguish
-between pods that it created or deleted and pods that another person or process created or
-deleted. This allows the ReplicaSet to be replaced without affecting the running pods.
+In this manner, a ReplicaSet can own a non-homogenous set of Pods
 
-The `.spec.template.metadata.labels` must match the `.spec.selector`, or it will
-be rejected by the API.
+## Writing a ReplicaSet Manifest
 
+As with all other Kubernetes API objects, a ReplicaSet needs the `apiVersion`, `kind`, and `metadata` fields.
+For ReplicaSets, the kind is always just ReplicaSet.
 In Kubernetes 1.9 the API version `apps/v1` on the ReplicaSet kind is the current version and is enabled by default. The API version `apps/v1beta2` is deprecated.
+Please refer to the first lines of the `frontend.yaml` example for guidance.
 
-Also you should not normally create any pods whose labels match this selector, either directly, with 
-another ReplicaSet, or with another controller such as a Deployment. If you do so, the ReplicaSet thinks that it 
-created the other pods. Kubernetes does not stop you from doing this.
+A ReplicaSet also needs a [`.spec` section](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status).
+
+### Pod Template
 
-If you do end up with multiple controllers that have overlapping selectors, you
-will have to manage the deletion yourself.
+The `.spec.template` is a [pod template](/docs/concepts/workloads/pods/pod-overview/#pod-templates) which also requires
+to have labels in place. In our `frontend.yaml` example we had one label: `tier: frontend`.
+Be careful not to overlap with the selectors of other controllers, lest they try to adopt this pod.
 
-### Labels on a ReplicaSet
+For the template's [restart policy](/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) field,
+`.spec.template.spec.restartPolicy`, the only allowed  value for  is `Always`, which is the default.
 
-The ReplicaSet can itself have labels (`.metadata.labels`).  Typically, you
-would set these the same as the `.spec.template.metadata.labels`.  However, they are allowed to be
-different, and the `.metadata.labels` do not affect the behavior of the ReplicaSet.
+### Pod Selector
+
+The `.spec.selector` field is a [label selector](/docs/concepts/overview/working-with-objects/labels/). As discussed
+[earlier](#how-a-replicaset-works) these are the labels used to identify potential pods to acquire. In our
+`frontend.yaml` example, the selector was:
+```shell
+matchLabels:
+	tier: frontend
+```
+
+The `.spec.template.metadata.labels` must match the `.spec.selector`, or it will
```

Comment:
It's not clear what object we're talking about here.
Suggestion: In the ReplicaSet, `.spec.template.metadata.labels` must match `spec.selector`, or ...